akka-http-core/src/main/scala/akka/http/javadsl/settings/ServerSettings.scala

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (C) 2016 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.http.javadsl.settings
+
+import java.util.{ Optional, Random }
+
+import akka.http.impl.settings.ServerSettingsImpl
+import akka.http.javadsl.model.headers.Host
+import akka.http.javadsl.model.headers.Server
+import akka.io.Inet.SocketOption
+import akka.http.impl.util.JavaMapping.Implicits._
+import scala.collection.JavaConverters._
+import scala.compat.java8.OptionConverters._
+
+import scala.concurrent.duration.{ Duration, FiniteDuration }
+
+abstract class ServerSettings {
+  def getServerHeader: Optional[Server]
+  def getTimeouts: ServerSettings.Timeouts
+  def getMaxConnections: Int
+  def getPipeliningLimit: Int
+  def getRemoteAddressHeader: Boolean
+  def getRawRequestUriHeader: Boolean
+  def getTransparentHeadRequests: Boolean
+  def getVerboseErrorMessages: Boolean
+  def getResponseHeaderSizeHint: Int
+  def getBacklog: Int
+  def getSocketOptions: java.lang.Iterable[SocketOption]
+  def getDefaultHostHeader: Host
+  def getWebsocketRandomFactory: java.util.function.Supplier[Random]
+  def getParserSettings: ParserSettings
+
+  // ---
+
+  def withServerHeader(newValue: Optional[Server]): ServerSettings = self.copy(serverHeader = newValue.asScala.map(_.asScala))
+  def withTimeouts(newValue: ServerSettings.Timeouts): ServerSettings = self.copy(timeouts = newValue.asScala)
+  def withMaxConnections(newValue: Int): ServerSettings = self.copy(maxConnections = newValue)
+  def withPipeliningLimit(newValue: Int): ServerSettings = self.copy(pipeliningLimit = newValue)
+  def withRemoteAddressHeader(newValue: Boolean): ServerSettings = self.copy(remoteAddressHeader = newValue)
+  def withRawRequestUriHeader(newValue: Boolean): ServerSettings = self.copy(rawRequestUriHeader = newValue)
+  def withTransparentHeadRequests(newValue: Boolean): ServerSettings = self.copy(transparentHeadRequests = newValue)
+  def withVerboseErrorMessages(newValue: Boolean): ServerSettings = self.copy(verboseErrorMessages = newValue)
+  def withResponseHeaderSizeHint(newValue: Int): ServerSettings = self.copy(responseHeaderSizeHint = newValue)
+  def withBacklog(newValue: Int): ServerSettings = self.copy(backlog = newValue)
+  def withSocketOptions(newValue: java.lang.Iterable[SocketOption]): ServerSettings = self.copy(socketOptions = newValue.asScala.toList)
+  def withDefaultHostHeader(newValue: Host): ServerSettings = self.copy(defaultHostHeader = newValue.asScala)
+  def withParserSettings(newValue: ParserSettings): ServerSettings = self.copy(parserSettings = newValue.asScala)
+  def withWebsocketRandomFactory(newValue: java.util.function.Supplier[Random]): ServerSettings = self.copy(websocketRandomFactory = () ⇒ newValue.get())
+
+  /** INTERNAL API */
+  protected def self = this.asInstanceOf[ServerSettingsImpl]
```

Comment:
Does not compile with selftype because then `self` is not available in scaladsl `ServerSettings` where it also used :(. 

Trying out self type with javap shows no public stuff leaked